src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -292,6 +294,84 @@ public Block deserialize(final String s) {
 					}
 				}));
 		
+		if (Skript.classExists("org.bukkit.block.data.BlockData")) {
+			Classes.registerClass(new ClassInfo<>(BlockData.class, "blockdata")
+				.user("block ?datas?")
+				.name("Block Data")
+				.description("Block data is the detailed information about a block, referred to in Minecraft as BlockStates, " +
+					"allowing for the manipulation of different aspects of the block, including shape, waterlogging, direction the block is facing, " +
+					"and so much more. Information regarding each block's optional data can be found on Minecraft's Wiki. Find the block you're " +
+					"looking for and scroll down to 'Block States'. Different states must be separated by a semicolon (see examples). " +
+					"The 'minecraft:' namespace is optional, as well as are underscores.")
+				.examples("set block at player to campfire[lit=false]",
+					"set target block of player to oak stairs[facing=north;waterlogged=true]",
+					"set block at player to grass_block[snowy=true]",
+					"set loop-block to minecraft:chest[facing=north]",
+					"set block above player to oak_log[axis=y]",
+					"set target block of player to minecraft:oak_leaves[distance=2;persistent=false]")
+				.after("itemtype")
+				.requiredPlugins("Minecraft 1.13+")
+				.since("INSERT VERSION")
+				.parser(new Parser<BlockData>() {
+					
+					@Nullable
+					@Override
+					public BlockData parse(String s, ParseContext context) {
+						return BlockUtils.createBlockData(s);
+					}
+					
+					@Override
+					public String toString(BlockData o, int flags) {
+						return o.getAsString().replace(",", ";");
+					}
+					
+					@Override
+					public String toVariableNameString(BlockData o) {
+						return "blockdata:" + o.getAsString();
+					}
+					
+					@Override
+					public String getVariableNamePattern() {
+						return "blockdata:.+";
+					}
+				})
+				.serializer(new Serializer<BlockData>() {
+					
```

Comment:
Same here with the space >:)

Reply:
Ok space being removed, will see in next commit!